types/types.go

```diff
@@ -331,6 +331,14 @@ type ImageDestination interface {
 	Commit(ctx context.Context, unparsedToplevel UnparsedImage) error
 }
 
+type ImageDestinationReusableWithRef interface {
+	ImageDestination
+
+	// TryReusingBlobWithRef checks whether the destination can efficiently reuse a blob, and if so, applies it to the current destination.
+	// This allows the destination to search the blob contents with taking the image reference into consideration.
+	TryReusingBlobWithRef(ctx context.Context, blobinfo BlobInfo, ref reference.Named) (bool, BlobInfo, error)
```

Comment:
See elsewhere about callers calling only one of the two `TryReusing*` methods.

Reply:
Fixed it by making `TryReusingBlobWithSrcRef` be a superset of `TryReusingBlob`.